discovery-provider/integration_tests/tasks/test_index_tastemaker_notification.py

```diff
@@ -0,0 +1,207 @@
+
+
+def test_index_tastemaker_notification_no_trending(app):
+    with app.app_context():
+        db = get_db()
+
+        # Add some users to the db so we have blocks
+        entities = {
+            "tracks": [{"track_id": i, "owner_id": 3} for i in range(5)],
+        }
+        populate_mock_db(db, entities)
+
+        index_tastemaker_notifications(
+            db=db,
+            top_trending_tracks=[],
+            tastemaker_notification_threshold=2,
+        )
+        with db.scoped_session() as session:
+            notifications = (
+                session.query(Notification)
+                .filter(Notification.type == "tastemaker")
+                .order_by(asc(Notification.specifier))
+                .all()
+            )
+            assert len(notifications) == 0
+
+
+def test_index_tastemaker_notification_no_tastemakers(app):
+    with app.app_context():
+        db = get_db()
+
+        # Add some users to the db so we have blocks
+        entities = {
+            "tracks": [{"track_id": i, "owner_id": 3} for i in range(5)],
+            "reposts": [
+                {
+                    "user_id": 1,
+                    "repost_item_id": 1,
+                    "repost_type": "track",
+                    "is_delete": False,
+                }
+            ],
+            "saves": [
+                {
+                    "user_id": 1,
+                    "save_item_id": 1,
+                    "save_type": "track",
+                    "is_delete": True,
+                },
+                {
+                    "user_id": 3,
+                    "save_item_id": 3,
+                    "save_type": "track",
+                    "is_current": False,
+                },
+                {
+                    "user_id": 4,
+                    "save_item_id": 3,
+                    "save_type": "track",
+                    "is_current": False,
+                },
+                {
+                    "user_id": 2,
+                    "save_item_id": 1,
+                    "save_type": "track",
+                    "is_current": False,
+                    "is_delete": True,
+                },
+            ],
+            "users": [{"user_id": i} for i in range(10)],
+        }
+        populate_mock_db(db, entities)
+
+        index_tastemaker_notifications(
+            db=db,
+            top_trending_tracks=[],
+            tastemaker_notification_threshold=2,
+        )
+        with db.scoped_session() as session:
+            notifications = (
+                session.query(Notification)
+                .filter(Notification.type == "tastemaker")
+                .order_by(asc(Notification.specifier))
+                .all()
+            )
+            assert len(notifications) == 0
+
+
+def test_index_tastemaker_notification_sends_one_notif_for_both_fav_and_repost(app):
+    with app.app_context():
+        db = get_db()
+
+        # Add some users to the db so we have blocks
```

Comment:
im starting to think we dont need this comment, cause it's clear its our mock data?

Reply:
yea great point lol, i think i copied this from an existing test so i will remove